packages/next-codemod/transforms/lib/async-request-api/utils.ts

```diff
@@ -492,3 +492,92 @@ export function getVariableDeclaratorId(
   }
   return undefined
 }
+
+export function findFunctionBody(path: ASTPath<FunctionScope>): null | any[] {
+  let functionBody = path.node.body
+  if (functionBody && functionBody.type === 'BlockStatement') {
+    return functionBody.body
+  }
+  return null
+}
+
+export const isReactHookLikeName = (name: string) =>
+  name.startsWith('use') && name[3] === name[3].toUpperCase()
+
+export function containsReactHooksCallExpressions(
+  path: ASTPath<FunctionScope>,
+  j: API['jscodeshift']
+) {
+  const hasReactHooks =
+    j(path)
+      .find(j.CallExpression)
+      .filter((callPath) => {
+        // use()
+        const isUseCall =
+          j.Identifier.check(callPath.value.callee) &&
+          isReactHookLikeName(callPath.value.callee.name)
+        // React.use()
+        const isReactUseCall =
+          j.MemberExpression.check(callPath.value.callee) &&
+          j.Identifier.check(callPath.value.callee.object) &&
+          callPath.value.callee.object.name === 'React' &&
+          j.Identifier.check(callPath.value.callee.property) &&
+          callPath.value.callee.property.name === 'use'
+        return isUseCall || isReactUseCall
+      })
+      .size() > 0
+  return hasReactHooks
+}
```

Comment:
This is just checking for `use()` but not arbitrary hooks. It needs to check for any hooks i.e. `use*` or `use` and if member expression, the object needs to be PascalCase.